rasterio/_warp.pyx

```diff
@@ -487,3 +491,60 @@ def _reproject(
         if dtypes.is_ndarray(source):
             if hdsin != NULL:
                 _gdal.GDALClose(hdsin)
+
+
+def _calculate_default_transform(
+        src_crs, dst_crs, width, height, left, bottom, right, top,
+        resolution=None, densify_pts=21):
```

Comment:
If `_calculate_default_transform()` is to replace `calculate_default_transform()` it will need at least a `**kwargs` for backwards compatibility.